### ISSUE TITLE
pyenv 2.3.0 (or latest master) init on Linux/Fedora 35 hanging/spawning many bash subprocesses when connecting via ssh

### ISSUE THREAD

**Issue by stevemuskiewicz** [pyenv/pyenv#2354]
### Description
- [X] Platform information: Fedora 35
- [X] OS architecture: x86_64
- [X] pyenv version: 2.3.0
- [X] Python version: 3.10.4
- [X] C Compiler information:  11.3.1
- [X] Please attach the debug trace of the failing command as a gist: [pyenv_init_debug.log](https://gist.github.com/stevemuskiewicz/11f7450d09d44c71f00b7c741bb9d877)

I recently did `pyenv update` to update from pyenv 2.2.5 to the latest `master`.  pyenv seems to work perfectly fine from my "local" Linux/Fedora 35 system (where I am logged directly - VNC/X session), but when I `ssh` to another Fedora 35 system (which uses my same home directory shared via NFS) _or_ to the same system that works locally, my shell init is hanging. 

As this is shell hang is occurring, if I do a check with `ps -ef`, I find what seems to be `pyenv` spawning a bunch of `bash` processes
 (partial capture of relevant output to illustrate):
```steve     353449  353401  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353497  353449  1 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353545  353497  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353593  353545  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353641  353593  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353689  353641  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353737  353689  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353785  353737  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353833  353785  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353881  353833  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353929  353881  0 10:35 pts/0    00:00:00 bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"
steve     353956  353929  0 10:35 pts/0    00:00:00 bash /home/steve/.pyenv/libexec/pyenv-init -
steve     353971  353956  0 10:35 pts/0    00:00:00 bash /home/steve/.pyenv/libexec/pyenv-commands --sh
steve     353972  353971  0 10:35 pts/0    00:00:00 bash /home/steve/.pyenv/libexec/pyenv-commands --sh
steve     353973  353971  0 10:35 pts/0    00:00:00 sort
steve     353974  353971  0 10:35 pts/0    00:00:00 uniq
steve     353975  330064  0 10:35 pts/1    00:00:00 ps -ef
```

I checked the [README](https://github.com/pyenv/pyenv#set-up-your-shell-environment-for-pyenv) and noticed that I still had the `eval "$(pyenv init --path)"` in my `~/.bashrc` so I tried removing it (as well as disabling the virtualenv plugin init) but it didn't appear to make any difference.  With these attempts the `pyenv` relevant portions of my `~/.bashrc` look like:
```
# .bashrc

# Source global definitions
if [ -f /etc/bashrc ]; then
        . /etc/bashrc
fi

path_append() {
  for ARG in "$@" ; do
    if [ -d "$ARG" ] && [[ ":$PATH:" != *":$ARG:"* ]]; then
        PATH="${PATH:+"$PATH:"}$ARG"
    fi
  done
}

path_append ${HOME}/.local/bin ${HOME}/bin
export PATH

export PYENV_ROOT="$HOME/.pyenv"
pathappend "${PYENV_ROOT}/bin"

  if command -v pyenv >/dev/null 2>&1 ; then
     eval "$(pyenv init -)"
     #eval "$(pyenv virtualenv-init -)"
  fi
```

If I revert `pyenv` to the v2.2.5 tag then the problem goes away and everything works as expected.  I tried using pyenv `v2.3.0` and the problem comes back so it would seem to be some kind of change as of v2.3.0 but not sure if it is specific to `bash` or Fedora 35 (running with the latest updates), or access via `ssh` or some combination of all of these.   I _think_ the fact that it doesn't work on my "local" system when I connect via `ssh` probably points to `ssh` as somehow being a culprit/related to the issue (and at the same time rules out NFS since it is acting as the NFS server for my home directory to the other Fedora systems that exhibit the same issue), but not entirely sure about this.

Other things I've done/tried:
* Did a `. ~/.bashrc` from the (working) "local" machine...in this case the pyenv init works fine regardless of version.
* I did add a `PYENV_DEBUG=1` and a capture of the `eval "$(pyenv init -)"` output, attaching that output as a gist.
* Checked `~/.bash_profile` to make sure I'm not doing anything with `PATH` assignments
* Did Ctrl-C during the shell hang to see what the state of the shell is... seems to give me a completely empty `PATH` variable and essentially an unusable shell session.
* Recompiled the dynamic bash extension (when `pyenv` is checked out to v2.3.0), ie. `cd ~/.pyenv && src/configure && make -C src`
* Removed the dynamic bash extension (`$HOME/.pyenv/libexec/pyenv-realpath.dylib`) entirely and retried - still hit the same hang issue

Not sure if I am missing or doing something incorrectly in `.bashrc` setup or if there is a real problem happening here.  Happy to try to debug further if someone could guide me as to what else to try.  I don't currently have any other non-Fedora 35 Linux VMs at my disposal but I could try this with a CentOS stream or Rockylinux VM if necessary to help isolate/rule out anything that might be Fedora-specific.

**Comment by stevemuskiewicz** [comment]
Also note that if I leave the system in the "hang" state and check `ps -ef` output, the number of processes for the same `bash` command as I initially observed just continues to increase!

This one:
```bash -ec IFS=:; paths=($PATH); for i in ${!paths[@]}; do if [[ ${paths[i]} == "/home/steve/.pyenv/shims" ]]; then unset 'paths[i]'; fi; done; echo "${paths[*]}"```

**Comment by native-api** [comment]
Do you by any chance have `BASH_ENV` set in your SSH environment? See the warning at the end of https://github.com/pyenv/pyenv#set-up-your-shell-environment-for-pyenv

**Comment by stevemuskiewicz** [comment]
@native-api thanks for the pointer...naturally I think I stopped scrolling right _before_ that section when reviewing the README 🤦 

I don't see `BASH_ENV` set in my SSH environment but looking thru the discussion thread referenced in the wiki, I see how having the init in `.bashrc` can possibly end up in a loop and it especially seems to be more of an issue with Redhat-ish systems.  I tried adding a conditional check on `[[ -n "$PS1" ]]` before the `eval "$(pyenv init -)"` in my `.bashrc` and with that change I am able to work around the hang issue when using pyenv 2.3.0!

Seems a little odd that I didn't need this for prior versions but it's a good enough workaround for me - thanks again!
